### PR TITLE
[RFC][DNM]Bluetooth: controller: Refined architecture

### PR DESCRIPTION
**Pull request by cvinayak**
Preview of the refined BLE controller architecture to solicit comments, improve maturity and enable early contributions.

Salient features:
- Enhanced radio event management in unreserved timespace.
- Stashing of pre-emptee radio event and pop out after pre-empter event.
- Close interleaving of advertising event inside continuous scanning window.

Supported states: legacy Advertising and Scanning.

Kconfig option to use:
CONFIG_BT_PERIPHERAL=n
CONFIG_BT_CENTRAL=n
CONFIG_BT_HCI_MESH_EXT=n
CONFIG_BT_LL_SW_SPLIT=y

Disclaimer: Implementation under the old architecture will fail to compile.

### CODE REVIEW

**Comment by cvinayak** [comment]
FYI, @pskhansen @aescolar-ot

File: subsys/bluetooth/shell/hci.c [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+		cp->own_addr_type = 0x01;
+		memset(&cp->random_addr, 0, sizeof(bt_addr_t));
+		cp->ch_map = 0x07;
+		cp->tx_power = 0;
+		cp->min_tx_delay = 0;
+		cp->max_tx_delay = 0x32;
+		cp->retx_count = 0x07;
+		cp->retx_interval = 0x00;
+		cp->scan_delay = 0x0a;
+		cp->scan_duration = 0x0064;
```
Comment (jhedberg) [inline subsys/bluetooth/shell/hci.c]: Missing `sys_cpu_to_le16()`?

**Comment by cvinayak** [comment]
New PR with further progress in refactoring will be submitted later.

**Comment by paulgavrikov** [comment]
@cvinayak I finally tested it with Mesh. Indeed I no longer see skipped Scan Windows and the swithcing time between two Windows is extremely small (80ns) - is it actually switching?
Also I have noticed that sometimes the Window does not close and the radio gets stuck in RX mode.

**Comment by cvinayak** [comment]
@paulgavrikov The switching time should be around 80 us (microseconds), are you really seeing 80 nanoseconds?

Do you have GPIO pins trace when you see radio stuck in RX mode? I have not personally seen the issue you describe, but GPIO trace can enlighten me of any bugs.

**Comment by paulgavrikov** [comment]
@cvinayak I should have zoomed in a bit better: between 2 Windows the Slave Pin is 20-80ns low + 16.6us high + 20-80ns low nothing. But 16us is also way too fast considering that the nRF needs 40us to ramp up. Am I watchign the wrong Pin?

It's hard to capture the exact moment it get's stuck but basically Slave & XTAL are high. LED3 remains off. When I try to send a Mesh message I get error -11.
Hope that helps - if you want me to log any other pins let me know.

**Comment by paulgavrikov** [comment]
@jhedberg I am mocking provisioning with `bt_mesh_provision(...)`  but sending works for a while until the nodes radio starts to get stuck. So I don't think it is related to provisioning.

**Comment by jhedberg** [comment]
@paulgavrikov I can't really find any other place that could return -EAGAIN, except perhaps if this gets propagated up all the way from the HCI driver (the controller). So is it `bt_mesh_model_send()` that's returning this or some other API?

**Pull request closed without merging**